File: test_phone_otp.js
// Node.js script to test Phone.email OTP Integration
// Run with: node test_phone_otp.js

const https = require('https');

const clientId = '16687983578815655151';
const apiKey = 'I1WSXNX52SESBCdtjbXIV8TjKhlQ8Qrf';
const phoneNumber = '+919811226924';

console.log('🚀 Testing Phone.email OTP Integration\n');
console.log('📱 Sending OTP to:', phoneNumber);
console.log('🔑 Using Client ID:', clientId);
console.log('');

const data = JSON.stringify({
  phone_number: phoneNumber
});

const options = {
  hostname: 'api.phone.email',
  port: 443,
  path: '/auth/v1/otp',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length,
    'X-Client-Id': clientId,
    'X-API-Key': apiKey
  }
};

const req = https.request(options, (res) => {
  console.log('📡 Response Status:', res.statusCode);
  console.log('');

  let responseBody = '';

  res.on('data', (chunk) => {
    responseBody += chunk;
  });

  res.on('end', () => {
    console.log('📄 Response Body:', responseBody);
    console.log('');

    if (res.statusCode === 200 || res.statusCode === 201) {
      try {
        const response = JSON.parse(responseBody);
        console.log('✅ SUCCESS! OTP sent successfully');
        console.log('📨 Session ID:', response.session_id || 'N/A');
        console.log('⏰ Expires in:', response.expires_in || 'N/A', 'seconds');
        console.log('');
        console.log('💡 Check your phone for the OTP code!');
      } catch (e) {
        console.log('✅ OTP sent (response parsing failed)');
      }
    } else {
      console.log('❌ FAILED to send OTP');
      console.log('Error:', responseBody);
    }
  });
});

req.on('error', (error) => {
  console.error('❌ ERROR:', error.message);
});

req.write(data);
req.end();

File: test_phone_otp.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

/// Test script to send OTP using Phone.email API
///
/// Client ID: 16687983578815655151
/// API Key: I1WSXNX52SESBCdtjbXIV8TjKhlQ8Qrf
/// Test Phone: +919811226924

void main() async {
  print('🚀 Testing Phone.email OTP Integration\n');

  const String clientId = '16687983578815655151';
  const String apiKey = 'I1WSXNX52SESBCdtjbXIV8TjKhlQ8Qrf';
  const String phoneNumber = '+919811226924';

  print('📱 Sending OTP to: $phoneNumber');
  print('🔑 Using Client ID: $clientId\n');

  try {
    // Send OTP request
    final response = await http.post(
      Uri.parse('https://api.phone.email/auth/v1/otp'),
      headers: {
        'Content-Type': 'application/json',
        'X-Client-Id': clientId,
        'X-API-Key': apiKey,
      },
      body: jsonEncode({'phone_number': phoneNumber}),
    );

    print('📡 Response Status: ${response.statusCode}');
    print('📄 Response Body: ${response.body}\n');

    if (response.statusCode == 200 || response.statusCode == 201) {
      final data = jsonDecode(response.body);
      print('✅ SUCCESS! OTP sent successfully');
      print('📨 Session ID: ${data['session_id'] ?? 'N/A'}');
      print('⏰ Expires in: ${data['expires_in'] ?? 'N/A'} seconds');
      print('\n💡 Check your phone for the OTP code!');
    } else {
      print('❌ FAILED to send OTP');
      print('Error: ${response.body}');
    }
  } catch (e) {
    print('❌ ERROR: $e');
  }
}
